src/control/lib/control/system.go

```diff
@@ -375,8 +382,13 @@ func rpcToRanks(ctx context.Context, rpcClient UnaryInvoker, req *RanksReq) (*Ra
 
 	rr := new(RanksResp)
 	for _, hostResp := range ur.Responses {
-		if hostResp.Error != nil {
-			if err := rr.addHostError(hostResp.Addr, hostResp.Error); err != nil {
+		hostErr := hostResp.Error
+		if hostErr != nil {
+			f, ok := errors.Cause(hostErr).(*fault.Fault)
+			if ok && f != nil {
+				hostErr = errors.New(f.Description)
+			}
```

Comment:
What's the intent behind this? The point of faults is that they contain more information than a simple error string. If the goal is to make printing the HostErrorsMap less verbose, I'd prefer to adjust the pretty printer rather than stripping out the fault details. Please remember that there may be other consumers for this information than just dmg...

Reply:
the problem is that when it's unpacked the error loses its form and is delegated to a string representation ( https://github.com/daos-stack/daos/pull/2701/files/96263bbec61defd1e160981332e5e6a63b558868#diff-0391e795d97fbda8de2f51693275526fR100 ) so if I want to avoid printing the verbose form , including the code which doesn't mean much to a user, I have to extract it here to avoid it being munged into the HostErrorsMap, losing the error type.

Reply:
Ah, right. Hmm. This was an oversight on my part, then. I'm kind of regretting taking the easy approach and using a string representation as the key. Something to revisit in the future perhaps.